### DISCUSSION TITLE
Automation - Looping over Create Row Action not possible

### DISCUSSION THREAD

**Discussion by VenSlazer** [Budibase/budibase#9821] [Help] [answered]
Hey,

I'm trying to use the Create Row Action with looping enabled. As soon as I want to test the automation, it get's stuck and throws the following error: 

`Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'filter')
    at X4t.n.$$.update (index.c2046c07.js:948:43146)
    at Wae (index.c2046c07.js:4:2669)
    at uM (index.c2046c07.js:4:2382)`

The only thing I can do is refresh the screen.

Here is how the automation looks like:
<img width="488" alt="Bildschirm­foto 2023-02-28 um 07 55 57" src="https://user-images.githubusercontent.com/77432946/221777517-11ec152f-baa5-4dc5-b24d-68b6bfe8721e.png">
<img width="488" alt="Bildschirm­foto 2023-02-28 um 07 56 23" src="https://user-images.githubusercontent.com/77432946/221777547-f9ab733d-9333-4874-b292-c048232c7d20.png">
<img width="490" alt="Bildschirm­foto 2023-02-28 um 07 56 38" src="https://user-images.githubusercontent.com/77432946/221777556-3fcf06da-f350-47ec-b944-63c79f6df228.png">

So I'm querying the data in step one and use it as a loop for step two/three. 
Is there a workaround, so I do not run into the error?

**Comment by PClmnt** [comment]
@VenSlazer Does the error persist whenever you remove the ∞ symbol from the limit field?

**Reply by VenSlazer** [reply]
Yes, I changed it to 20 from ∞, but I still get the same error.

**Reply by melohagan** [reply]
@VenSlazer Do you have any filters defined? If so could you share a screenshot of them please.

**Reply by VenSlazer** [reply]
I tried it with and without filters => same result. 
Here is an example of a filter I tried:
<img width="1471" alt="Bildschirm­foto 2023-03-02 um 08 22 06" src="https://user-images.githubusercontent.com/77432946/222358905-070d1705-6ba2-4f2f-a6f1-18ee262d6181.png">

**Reply by melohagan** [reply]
Thanks @VenSlazer 

Would you be able to send your app export to support@budibase.com ? I can then take a closer look.

**Reply by VenSlazer** [reply]
I'd like to, but your Email-Server does not accept the .tar.gz export I'm trying to attach to the email.

**Reply by melohagan** [reply]
Would you be able to email me a link to a Google Drive for example?

**Reply by VenSlazer** [reply]
I just sent it to the support@budibase.com email address.

**Reply by VenSlazer** [reply]
Ok, so I just installed the newly released version 2.3.21 via Docker. 
When I now try to run the automation, it's not getting stuck anymore, but I receive an error toast message that says "Error testing automation". Would be great to know what triggerd this error, but there is no log information given. 

@melohagan did you receive my Email and did it work with the download link?

**Reply by melohagan** [reply]
Hey @VenSlazer 

I did get your email and export, thank you. This new issue you're seeing appears to be related to this: https://github.com/Budibase/budibase/discussions/9903

I'm taking a look now.

**Reply by VenSlazer** [reply]
Nice, thank you 👍

**Comment by melohagan** [comment, marked as answer]
Hey @VenSlazer 

This appears to be a bug which I have raised here: https://github.com/Budibase/budibase/issues/9916

**Reply by VenSlazer** [reply]
Hey @melohagan ,

so I've updated to 2.4.4 and tried to stringify the result of the query-step in the looping step (like in #9903), but I still receive the Error code. So at least for me it is not working.

**Reply by melohagan** [reply]
You may need to wait for the bug fix then: https://github.com/Budibase/budibase/pull/9924

When that is merged and deployed that should hopefully resolve the issues you're seeing!
